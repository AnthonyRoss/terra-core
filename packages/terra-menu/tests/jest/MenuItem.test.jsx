import React from 'react';
import MenuItem from '../../src/MenuItem';

// Snapshot Tests
it('should render with className', () => {
  const item = shallow(<MenuItem className="textClass" />);
  expect(item).toMatchSnapshot();
});

it('should render with text', () => {
  const item = shallow(<MenuItem text="Text" />);
  expect(item).toMatchSnapshot();
});

it('should render with selected', () => {
  const item = shallow(<MenuItem isSelected />);
  expect(item).toMatchSnapshot();
});

it('should render with isSelectable', () => {
  const item = shallow(<MenuItem isSelectable />);
  expect(item).toMatchSnapshot();
});

it('should render as disabled', () => {
  const item = shallow(<MenuItem isDisabled />);
  expect(item).toMatchSnapshot();
});
