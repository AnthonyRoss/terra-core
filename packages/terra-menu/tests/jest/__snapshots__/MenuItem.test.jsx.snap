// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as disabled 1`] = `
<li
  aria-disabled={true}
  className="item is-disabled"
  role="menuitem"
  tabIndex="-1"
>
  <div
    className="title"
  />
</li>
`;

exports[`should render with className 1`] = `
<li
  aria-disabled={false}
  className="item textClass"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <div
    className="title"
  />
</li>
`;

exports[`should render with isSelectable 1`] = `
<li
  aria-disabled={false}
  className="item"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="menuitemcheckbox"
  tabIndex="0"
>
  <div
    className="title"
  />
</li>
`;

exports[`should render with selected 1`] = `
<li
  aria-disabled={false}
  className="item"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <div
    className="title"
  />
</li>
`;

exports[`should render with text 1`] = `
<li
  aria-disabled={false}
  className="item"
  onClick={[Function]}
  onKeyDown={[Function]}
  role="menuitem"
  tabIndex="0"
>
  <div
    className="title"
  >
    Text
  </div>
</li>
`;
